File: Retos/Reto #4 - PRIMO, FIBONACCI Y PAR [Media]/javascript/pyramsd.js
function es_fibonacci(n){
    let a = 0;
    let b = 1;
    while (b < n){
        let varTemp = a;
        a = b;
        b = varTemp + b;
        if (b > n) break;
    }
     return b == n;
}

// console.log(es_fibonacci(9));

function es_parImpar(n){
    return n % 2 === 0;
}

// console.log(es_parImpar(4));

function es_primo(n){
    if (n < 2){
        return false;
    }
    for (let i = 2; i <= Math.sqrt(n); i++){
        if (n % i === 0){
            return false;
        }
    }
    return true;
}

// console.log(es_primo(4));

function respuestaFinal(num){
    let res_final = "";
    if (es_primo(num)){
        res_final += "es primo, ";
    }else{
        res_final += "no es primo, ";
    }
    if (es_fibonacci(num)){
        res_final += "es fibonacci, ";
    }else{
        res_final += "no es fibinacci, ";
    }
    if (es_parImpar(num)){
        res_final += "y es par";
    }else{
        res_final += "y no es par";
    }
    return `${num} ${res_final}`;
}

console.log(`${respuestaFinal(3)}`);

File: Retos/Reto #4 - PRIMO, FIBONACCI Y PAR [Media]/python/pyramsd.py
def es_fibinacci(n):
    a, b = 0, 1
    while b < n:
        varTemp = a
        a, b = b, varTemp + b
    return b == n

def es_ParImpar(n):
    if n % 2 == 0:
        return True
    return False

def es_primo(n):
    if n < 2:
        return False
    for i in range(2, n):
        if n % i == 0:
            return False
    return True

def respuestaFinal():
    res_final = ""
    if es_primo(num):
        res_final += "es primo, "
    else:
        res_final += "no es primo, "
    if es_fibinacci(num):
        res_final += "fibonacci, "
    else:
        res_final += "no es fibonacci, "
    if es_ParImpar(num):
        res_final += "y es par."
    else:
        res_final += "y es impar."
    return res_final

num = int(input("Numero: "))

print(f"{num} {respuestaFinal()}")
